lib/topic_creator.rb

```diff
@@ -27,6 +27,8 @@ def valid?
     # not sure where this should go
     if !@guardian.is_staff? && staff_only = DiscourseTagging.staff_only_tags(@opts[:tags])
       topic.errors[:base] << I18n.t("tags.staff_tag_disallowed", tag: staff_only.join(" "))
+    elsif !@guardian.can_create_tag? || !SiteSetting.tagging_enabled?
+      topic.errors[:base] << I18n.t("errors.messages.not_allowed", operation: "Tag Creation")
```

Comment:
We can't have this error cause "operation" is not localized, so you end up having a half localized string, also ideally we need a test here. 

A bit uneasy to have the error handling now in 2 spots due to this change.

Reply:
so should i remove operation and write complete error in locale?

Reply:
yes, but I am uneasy about having this handled in 2 spots, what error do you get now when you are missing a tag?

Reply:
hi i have done some changes please check.
i am getting "Tag creation not allowed " error when user can not create tags and site setting tagging is not enabled.

Reply:
hi @SamSaffron 
i have done some changes please review. i am getting "Tag creation not allowed" error when user don't have tag creation permission or site setting -tagging is not enabled.